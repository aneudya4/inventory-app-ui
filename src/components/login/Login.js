import React, { useCallback, useContext, useState } from 'react';
import { Redirect } from 'react-router-dom';
import app from '../../firebaseConfig';
import apiContext from '../../apiContext';
import { Link } from 'react-router-dom';
import './login.css';
const Login = ({ history }) => {
  const { user } = useContext(apiContext);
  const [loginError, setLoginError] = useState(false);

  const handleLogin = useCallback(
    async (e) => {
      e.preventDefault();
      const { email, password } = e.target.elements;
      try {
        await app
          .auth()
          .signInWithEmailAndPassword(email.value, password.value);

        history.push('/auth/dashboard/overview');
      } catch (error) {
        setLoginError(true);
      }
    },
    [history]
  );

  if (user) {
    return <Redirect to='/auth/dashboard/overview' />;
  }

  return (
    <div className='login-form form'>
      <form onSubmit={handleLogin}>
        {loginError && (
          <span className='validation-errors'>User Account Does Not Exist</span>
        )}
        <h3>Log In</h3>
        <label htmlFor='email'>
          Your Email*
          <input
            id='email'
            name='email'
            type='email'
            placeholder='jhonDoe@email.com'
          />
        </label>

        <label htmlFor='password'>
          Your Password*
          <input
            id='password'
            name='password'
            type='password'
            placeholder='your password'
          />
        </label>

        <button className='btn'>Login</button>
        <span>
          need an account ? <Link to='./register'>Register</Link>
        </span>
      </form>
    </div>
  );
};

export default Login;
