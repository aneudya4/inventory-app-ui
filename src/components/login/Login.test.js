import { shallow } from 'enzyme';

import Login from './Login';

describe('Test on the <Login/> Component', () => {
  const wrapper = shallow(<Login />);

  it('renders App without crashing', () => {
    expect(wrapper.find('form').exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
