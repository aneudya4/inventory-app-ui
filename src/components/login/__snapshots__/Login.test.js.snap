// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <Login/> Component renders App without crashing 1`] = `
<div
  className="login-form form"
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Log In
    </h3>
    <label
      htmlFor="email"
    >
      Your Email*
      <input
        autoComplete="off"
        id="email"
        name="email"
        placeholder="jhonDoe@email.com"
        required={true}
        type="email"
      />
    </label>
    <label
      htmlFor="password"
    >
      Your Password*
      <input
        autoComplete="off"
        id="password"
        name="password"
        placeholder="your password"
        required={true}
        type="password"
      />
    </label>
    <span
      className="btn"
      onClick={[Function]}
    >
      DEMO
    </span>
    <button
      className="btn"
    >
      Login
    </button>
    <span>
      need an account ? 
      <Link
        to="./register"
      >
        Register
      </Link>
    </span>
  </form>
</div>
`;
